src/ray/rpc/worker/core_worker_client_pool.cc

```diff
@@ -0,0 +1,47 @@
+#include "ray/rpc/worker/core_worker_client_pool.h"
+
+namespace ray {
+namespace rpc {
+
+optional<shared_ptr<CoreWorkerClientInterface>> CoreWorkerClientPool::GetByID(
+    ray::WorkerID id) {
+  absl::MutexLock lock(&mu_);
```

Comment:
Just using a reader lock?

Reply:
Overall, it's easier/defensive to use an everyday lock unless the lock contention is going to be a problem (ie, there's more in the function than a map access)